### PR TITLE
Experiences

### PR DESCRIPTION
**Pull request by DanielWilliamParsons**

### CODE REVIEW

File: src/utils/interfaces/Place.ts [hunk 1]
```diff
@@ -35,6 +35,9 @@ interface Experience {
   id: number;
   placeId: number;
   userId: number;
+  username: string;
+  dateVisited: Date;
+  etiquette: string;
```
Comment (kiw1meister) [inline src/utils/interfaces/Place.ts:40]: Can likely change this later, but might this use the etiquette ID as opposed to the label/string?

File: src/views/Experiences/ExperiencesComponents/ExperiencesComponent.vue [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    // and add to array only if not contained in the array
+    // A set is good for this because sets only accept unique values
+    const uniqueEtiquettes = new Set<string>();
+    place.details.experiences.forEach((experience) => {
+        uniqueEtiquettes.add(experience.etiquette);
+    });
+    const etiquetteTypes: string[] = Array.from(uniqueEtiquettes);
+    const selectedFilter = ref<string | "">("");
+
+    // Filter experiences based on experienceType
```
Comment (kiw1meister) [inline src/views/Experiences/ExperiencesComponents/ExperiencesComponent.vue:23]: Good use of computed() for filtering!

File: src/views/Experiences/ExperiencesComponents/ExperiencesComponent.vue [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    const place = usePlaceStore();
+
+    //use mock state data
+    place.useMock();    
+
+    const emit = defineEmits(['toggleAddExperience']);
+
+    // etiquettes must be taken from Place -> Experience -> Etiquette
+    // and add to array only if not contained in the array
+    // A set is good for this because sets only accept unique values
```
Comment (kiw1meister) [inline src/views/Experiences/ExperiencesComponents/ExperiencesComponent.vue:15]: Good thinking on using a set!

**Review by kiw1meister — APPROVED** [review]
Seems good! I will merge this

File: src/utils/interfaces/Place.ts (hunk 1 above)
Comment (DanielWilliamParsons) [reply src/utils/interfaces/Place.ts:40]: agreed

File: src/views/Experiences/ExperiencesComponents/ExperiencesComponent.vue (hunk 2 above)
Comment (DanielWilliamParsons) [reply src/views/Experiences/ExperiencesComponents/ExperiencesComponent.vue:23]: Thanks! Still learning Vue : )

File: src/views/Experiences/ExperiencesComponents/ExperiencesComponent.vue (hunk 3 above)
Comment (DanielWilliamParsons) [reply src/views/Experiences/ExperiencesComponents/ExperiencesComponent.vue:15]: Thanks!

**Comment by DanielWilliamParsons** [comment]
Thanks for the comments - I'll be more rigorous in my comments on yours from now on : )

**Pull request merged**